plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditorOutlinePage.java

```diff
@@ -191,12 +177,16 @@ public void selectionChanged(@NotNull SelectionChangedEvent event) {
     }
 
     private void scheduleRefresh() {
-        switch (this.refreshJob.getState()) {
-            case Job.WAITING, Job.SLEEPING -> this.refreshJob.cancel();
-        }
-        this.refreshJob.schedule(500);
+        UIUtils.asyncExec(new Runnable() {
+            @Override
+            public void run() {
+                if (!treeViewer.getTree().isDisposed()) {
+                    treeViewer.refresh();
+                }
+            }
+        });
```

Comment:
We're using Java 17, not Java 7.

```suggestion
        UIUtils.asyncExec(() -> {
            if (!treeViewer.getTree().isDisposed()) {
                treeViewer.refresh();
            }
        });
```

Reply:
Done